data-prepper-plugins/map-values-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/mapvalues/MapValues.java

```diff
@@ -0,0 +1,179 @@
+
+
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.mapvalues;
+
+import org.apache.commons.lang3.Range;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.EVENT;
+
+
+@DataPrepperPlugin(name = "map-values", pluginType = Processor.class, pluginConfigurationType = MapValuesConfig.class)
+public class MapValues extends AbstractProcessor<Record<Event>, Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MapValues.class);
+    private final ExpressionEvaluator expressionEvaluator;
+    private final MapValuesConfig config;
+    private final LinkedHashMap<Range<Integer>, String> RANGES;
+    private final Map<String, String> MAP;
+
+    private final HashMap<String, String> PATTERNS;
+    @DataPrepperPluginConstructor
+    public MapValues(PluginMetrics pluginMetrics, final MapValuesConfig mapValuesConfig, final ExpressionEvaluator expressionEvaluator) {
+        super(pluginMetrics);
+        this.config = mapValuesConfig;
+        this.expressionEvaluator = expressionEvaluator;
+        // LOG.info("Entered mapvalues constructor");
+        MAP = new HashMap<>();
+        RANGES = new LinkedHashMap<>();
+        PATTERNS = config.getPatterns();
+        processMapField(mapValuesConfig.getMap());
+        parseFile(mapValuesConfig.getFilePath());
+
+    }
+
+    private void processMapField(Map<String, String> map){
+        for(Map.Entry<String, String> mapEntry : map.entrySet()){
+            String key = mapEntry.getKey();
+            if(key.contains(",")){
```

Comment:
I think we should also allow something like `200-210,220-230`. I think it is an easy change. First let's split by ',' and then look for either individual entries or ranges.  Something like this -

```
String[] keys = mapEntry.getKey().split(",");
for (String key: keys) {
    if (key.contains("-")) {
         parseRange(key);
     } else {
         MAP.put(key, mapEntry.getValue());
    }

Reply:
Yes that seems a reasonable fucntionality. Will add that to the next commit.